### PR TITLE
Refactor TestController. Implement t.eval() (#241)

### PR DESCRIPTION
**Pull request by inikulin**
\cc @AlexanderMoskovkin @helen-dikareva @AndreyBelym 
Note that I've changed the way we define API methods to avoid code duplication.
@VasilyStrelyaev pls, check strings

### CODE REVIEW

File: src/api/test-controller.js [hunk 1]
```diff
@@ -98,86 +106,70 @@ export default class TestController {
...
-    typeText (selector, text, options) {
-        return this._typeTextImpl(selector, text, options);
+        return hybridFn();
     }
+}
 
-    selectText (selector, startPos, endPos) {
-        return this._selectTextImpl(selector, startPos, endPos);
-    }
+(function createAPIMethods () {
```
Comment (helen-dikareva) [inline src/api/test-controller.js]: May be better to do it in constructor (like `_createAPIMethodsList`)

Comment (inikulin) [inline src/api/test-controller.js]: Nope. We add methods to `prototype`, if we do it constructor we will add unnecessary memory overhead

**Comment by AlexanderMoskovkin** [comment]
lgtm

**Comment by AndreyBelym** [comment]
lgtm

**Comment by helen-dikareva** [comment]
ok. lgtm

**Comment by inikulin** [comment]
ping @VasilyStrelyaev

File: test/functional/fixtures/api/es-next/eval/test.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+var expect         = require('chai').expect;
+var parseUserAgent = require('useragent').parse;
+
+describe('[API] t.eval', function () {
+    it('Should execute anonymous hybrid function', function () {
```
Comment (VasilyStrelyaev) [inline test/functional/fixtures/api/es-next/eval/test.js]:
```
    it('Should execute an anonymous hybrid function', function () {
```

File: test/functional/fixtures/api/es-next/eval/test.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+        return runTests('./testcafe-fixtures/eval-test.js', 'Get UA', { shouldFail: true })
+            .catch(function (errs) {
+                assertUA(errs, 'chrome', 'Chrome');
+                assertUA(errs, 'ff', 'Firefox');
+                assertUA(errs, 'ie', 'IE');
+            });
+    });
+
+    it('Should execute anonymous hybrid function with dependencies', function () {
```
Comment (VasilyStrelyaev) [inline test/functional/fixtures/api/es-next/eval/test.js]:
```
    it('Should execute an anonymous hybrid function with dependencies', function () {
```

File: test/functional/fixtures/api/es-next/eval/test.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+        return runTests('./testcafe-fixtures/eval-test.js', 'Eval with dependencies');
+    });
+
+    it('Should have correct callsite if error occurs on instantiation', function () {
```
Comment (VasilyStrelyaev) [inline test/functional/fixtures/api/es-next/eval/test.js]:
```
    it('Should have the correct callsite if an error occurs on instantiation', function () {
```

File: test/functional/fixtures/api/es-next/eval/test.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        return runTests('./testcafe-fixtures/eval-test.js', 'Error on instantiation', { shouldFail: true })
+            .catch(function (errs) {
+                expect(errs[0]).contains('Client code is expected to be specified as a function, but "string" was passed.');
+                expect(errs[0]).contains("> 21 |    await t.eval('42');");
+            });
+    });
+
+    it('Should have correct callsite if error occurs during execution', function () {
```
Comment (VasilyStrelyaev) [inline test/functional/fixtures/api/es-next/eval/test.js]:
```
    it('Should have the correct callsite if an error occurs during execution', function () {
```

**Comment by VasilyStrelyaev** [comment]
\r-

**Comment by VasilyStrelyaev** [comment]
@inikulin BTW, is *anonymous hybrid function* the official term we should now use for `t.eval()` ?

**Comment by inikulin** [comment]
@VasilyStrelyaev To be more precise it's self-invocked anonymous hybrid function without args. It has the same machinery as hybrid function (since it's uses them internally). But I think we should avoid using `anonymous hybrid function` term in docs because such definition might be confusing. On the one hand it's a shorthand for Hybrids, on the other hand it's improved eval which exists in other frameworks and might be more familiar for the users

**Comment by VasilyStrelyaev** [comment]
@inikulin yeah, that's exactly what I was thinking

**Comment by inikulin** [comment]
FPR

**Comment by VasilyStrelyaev** [comment]
lgtm

**Comment by inikulin** [comment]
I guess it can be merged now

**Pull request merged**